students/BrianB/lesson07/html_render.py

```diff
@@ -0,0 +1,225 @@
+    tag = "body"
+
+
+class P(Element):
+    tag = "p"
+
+
+class Head(Element):
+    tag = "head"
+
+
+class Li(Element):
+    """
+    I was unable to follow the instructions beyond
+    mid-Step 4, this was borrowed from an alt source
+    """
+    tag = 'li'
+
+
+class Ul(Element):
+    """
+    I was unable to follow the instructions beyond
+    mid-Step 4, this was borrowed from an alt source
+    """
+    tag = 'ul'
+
+    def __init__(self, content=None, **kwargs):
+        """
+        I was unable to follow the instructions beyond
+        mid-Step 4, this was borrowed from an alt source
+        """
+        if content: raise TypeError
+
+        self.contents = []
+        self.attributes = kwargs
+
+
+class OneLineTag(Element):
+
+    def render(self, out_file):
+        # loop through the list of contents
+        out_file.write("<{}>".format(self.tag))
+        out_file.write(self.contents[0])
+        out_file.write("</{}>\n".format(self.tag))
+
+    def append(self, content):
+        raise NotImplementedError
+
+
+class H(OneLineTag):
+    """
+    I was unable to follow the instructions beyond
+    mid-Step 4, this was borrowed from an alt source
+    """
+    tag = 'h'
+
+    def __init__(self, level, content=None, **kwargs):
+        """
+        I was unable to follow the instructions beyond
+        mid-Step 4, this was borrowed from an alt source
+        """
+        try:
+            int(level)
+        except ValueError:
+            raise ValueError
+
+        self.tag = f'h{level}'
+        super().__init__(content, **kwargs)
+
+
+class Title(OneLineTag):
+    tag = "title"
+
+
+class A(OneLineTag):
+    """
+    I was unable to follow the instructions beyond
+    mid-Step 4, this was borrowed from an alt source
+    """
+    tag = 'a'
+
+    def __init__(self, link, content=None, **kwargs):
+        """
+        I was unable to follow the instructions beyond
+        mid-Step 4, this was borrowed from an alt source
+        """
+        if not (content and link): raise TypeError
+
+        kwargs['href'] = link
+        super().__init__(content, **kwargs)
+
+
+class SelfClosingTag(Element):
+    """
+    I was unable to follow the instructions beyond
+    mid-Step 4, this was borrowed from an alt source
+    """
+    def __init__(self, content=None, **kwargs):
+        """
+        I was unable to follow the instructions beyond
+        mid-Step 4, this was borrowed from an alt source
+        """
+        if content: raise TypeError
+        self.attributes = kwargs
+
+    def append(self, new_content):
+        """
+        I was unable to follow the instructions beyond
+        mid-Step 4, this was borrowed from an alt source
+        """
+        raise NotImplementedError
+
+    def render(self, out_file, indent="", ind_count=0):
+        """
+        I was unable to follow the instructions beyond
+        mid-Step 4, this was borrowed from an alt source
+        """
+        self.indent = indent * ind_count
```

Comment:
you don't need to set it in self.

and in this context, you don't want the ind_count multiplied by the indent passed in. If you are using a count, then it would be:

cur_indent = self.indent * ind_count

where self.indent is a class attribute of Element -- and does not change.